src/zjs_modules.c

```diff
@@ -95,16 +96,26 @@ static jerry_value_t native_require_handler(const jerry_value_t function_obj,
     for (int i = 0; i < modcount; i++) {
         module_t *mod = &zjs_modules_array[i];
         if (!strcmp(mod->name, module)) {
-            return jerry_acquire_value(mod->init());
+            // We only want one intance of each module at a time
+            if (mod->instance == 0) {
+                mod->instance = jerry_acquire_value(mod->init());
+            }
+            return mod->instance;
         }
     }
 
-    PRINT("MODULE: `%s'\n", module);
     return zjs_error("native_require_handler: module not found");
 }
 
 void zjs_modules_init()
 {
+    int modcount = sizeof(zjs_modules_array) / sizeof(module_t);
+
+    for (int i = 0; i < modcount; i++) {
+        module_t *mod = &zjs_modules_array[i];
+        mod->instance = 0;
```

Comment:
I'm 99% sure that since the modules array was created as static global data it is already initialized to zero. On one hand it doesn't hurt anything, but on the other hand we're trying to pinch bytes here! Can you verify this?